neo4j/transaction_with_context.go

```diff
@@ -70,26 +88,23 @@ func (tx *explicitTransaction) Run(ctx context.Context, cypher string, params ma
 	}
 	stream, err := tx.conn.RunTx(ctx, tx.txHandle, db.Command{Cypher: cypher, Params: params, FetchSize: tx.fetchSize})
 	if err != nil {
-		tx.err = err
-		tx.runFailed = true
+		tx.txState.onError(err)
```

Comment:
fixes `test_should_prevent_discard_after_tx_termination_on_run due` and `test_should_prevent_pull_after_tx_termination_on_run` by propagating errors